### DISCUSSION TITLE
Are there examples of external data collection modules written in languages that aren't Python or Go?

### DISCUSSION THREAD

**Discussion by rshen-tibco** [netdata/netdata#15972] [Q&A] [answered]
I know Netdata itself collects a lot of system metrics in C, and there are plugin orchestrators for modules written in Go or Python. But are there external plugins that collect in C and send to Netdata without a plugin orchestrator? Or any that use an external database but not one of the provided plugin orchestrators?

**Comment by Ferroin** [comment, marked as answer]
Well, we have a framework for shell scripts for data collection (found here in our source code: https://github.com/netdata/netdata/tree/master/collectors/charts.d.plugin), and we _used_ to have some collectors written in Node.JS (it was removed a while back, but you can see an old copy of the code here: https://github.com/netdata/netdata/tree/v1.34/collectors/node.d.plugin).

In general though, we don’t really have much in the way of ‘simple’ external plugins that don’t use our Python or Go frameworks or our internal C API any more.

**Reply by rshen-tibco** [reply]
Is there a way for external plugins to access the internal C API then? Just wondering since you say "anymore".

**Reply by rshen-tibco** [reply]
@Ferroin Sorry for the tag; I'm starting to wonder if GitHub notifies you of responses to your response because on the last few questions I've asked, I've gotten one reply and then nothing after that.

**Reply by Ferroin** [reply]
> Is there a way for external plugins to access the internal C API then? Just wondering since you say "anymore".

Not really. The internal C API is built around the assumption of in-process messaging.

> @Ferroin Sorry for the tag; I'm starting to wonder if GitHub notifies you of responses to your response because on the last few questions I've asked, I've gotten one reply and then nothing after that.

Yeah, for some reason GitHub doesn’t notify about replies to replies. It’s one of the ore significant limitations of the platform at the moment.

**Comment by andrewm4894** [comment]
I think this is an example in bash 

https://github.com/netdata/community/tree/main/collectors/charts.d.plugin/speedtest

And a node.js one 

https://github.com/netdata/community/tree/main/collectors/node.d.plugin/snmp

**Reply by rshen-tibco** [reply]
Thank you! GitHub wouldn't let me mark two answers, but I appreciate them both 👍
